### DISCUSSION TITLE
File won't open when writing attachment file stream to file

### DISCUSSION THREAD

**Discussion by Minisaints** [Jericho/StrongGrid#449] [Q&A]
Hey,

I'm trying to write the attachment stream to an actual file binary using a file stream but the file won't open. The file binary opened in notepad appears to resemble normal binary data too. 

Am I missing something?

![Broken_Image](https://user-images.githubusercontent.com/34611622/163633620-f22e9a2f-ae3c-4df7-af0f-1d3164c9b9dc.jpg)

```
Public Sub WriteToFile(ByVal stream As Stream, filePath As String)
            stream.Seek(0, SeekOrigin.Begin)

            Using fs = New FileStream(filePath, FileMode.OpenOrCreate)
                stream.CopyTo(fs)
            End Using
 End Sub

WriteToFile(attachment.Data, Path.Combine(filePath, $"{fileID}_{fileName}"))
```

**Comment by Jericho** [comment]
Here's how I save attachments files in my `c:\temp` folder:

```csharp
var parser = new WebhookParser();
var inboundEmail = await parser.ParseInboundEmailWebhookAsync(Request.Body).ConfigureAwait(false);
if (inboundEmail.Attachments != null)
{
	foreach (var attachment in inboundEmail.Attachments)
	{
		byte[] content = new byte[attachment.Data.Length];
		attachment.Data.Read(content, 0, (int)attachment.Data.Length);

		var filePath = System.IO.Path.Combine("C:\\Temp", attachment.FileName);
		System.IO.File.WriteAllBytes(filePath, content);
	}
}
```

Hopefully this helps

**Reply by Minisaints** [reply]
Sorry for my late reply and thank you for the snippet.

The previous code was passing through a memory stream to 'ParseInboundEmailWebhookAsync' created from the request body as UTF-8 instead of just copying the request stream object directly.

All working great now. 

Thanks again.
